handshake-layer.go

```diff
@@ -171,18 +216,125 @@ func (h *HandshakeLayer) sendAlert(err Alert) error {
 	return nil
 }
 
+func (h *HandshakeLayer) messageDelivered(seq uint32) {
```

Comment:
This method name could be more descriptive.  `noteMessageDelivered` or something about queue updating?